### PR TITLE
Delay progress spinner rendering

### LINKED ISSUE #3120: Mailbody flickering even when already loaded
When switching between mails in the mailview, the mailviewer will flicker each time, by showing a loading spinner for a frame before the content is attached, it looks pretty ugly. If the mailbody has been loaded already we should be able to avoid this somehow.

### PR DESCRIPTION
**Pull request by jowlo**
When displaying a mail we download and sanitize the mail. Until this is ready we show a progress spinner in the main body view of the email.

We also show the spinner while only sanitizing already downloaded data. This delays the rendering of the progress spinner by 300ms to reduce the flickering when skipping through mails at a quick rate.

fix #3120

### CODE REVIEW

**Comment by jowlo** [comment]
@johnbotris Could you test if this fixes the issues for you? I could only see the problem on some mails when building against prod.

File: src/mail/view/MailViewer.js [hunk 1]
```diff
@@ -264,6 +264,11 @@ export class MailViewer {
 				m.redraw()
 			})
 
+		// Delay the display of the progress spinner in main body view for a short time to suppress it when just sanitizing
+		this._delayProgressSpinner = true
+		new Promise(resolve => setTimeout(resolve, 300))
```
Comment (johnbotris) [inline src/mail/view/MailViewer.js:269]: should use `Promise.delay` or just pass the callback directly to `setTimeout` pls

**Comment by johnbotris** [comment]
* Sometimes the spinner doesn't show at all, even after the delay when the mail hasn't yet loaded
* 300ms is a bit long, it looks good when flicking between already loaded mails but feels a bit sluggish otherwise. Probably shouldn't need to be more than ~50ms or something..

**Review by johnbotris — CHANGES_REQUESTED** [review]

File: src/mail/view/MailViewer.js (hunk 1 above)
Comment (jowlo) [reply src/mail/view/MailViewer.js]: Sure, don't really know how it got this convoluted...

**Comment by jowlo** [comment]
I was thinking: Would it be helpful to actually make a component that displays the progress spinner with a little delay? Maybe make it slowly fading in after the delay? Could be helpful for cases like #3157 as well...

**Comment by johnbotris** [comment]
> I was thinking: Would it be helpful to actually make a component that displays the progress spinner with a little delay? Maybe make it slowly fading in after the delay? Could be helpful for cases like #3157 as well...

I think that would be a nice idea

**Comment by charlag** [comment]
What's the status of this now? Changes were requested and then some changes were made

**Comment by jowlo** [comment]
> What's the status of this now? Changes were requested and then some changes were made

Iirc it works and fixes the flckering but it does not make a generic component like a `DelayedSpinner`. But since #3157 is fixed differently anyways we might just do it here for now and extract when needed. Maybe @johnbotris  could do the review again?

**Comment by charlag** [comment]
@johnbotris pls check it

**Review by johnbotris — APPROVED** [review]

**Pull request merged**